build.rs

```diff
@@ -121,168 +93,142 @@ fn main() {
     }
-
-    if renew {
-        let args = [
-            "-o",                           // Overwrite files without prompting
-            sdk_zip_file.to_str().unwrap(), // The zip file
-            "-d",
-            sdk_output_dir.to_str().unwrap(), // The directory to extract to
-        ];
-        Command::new("unzip").args(args).status().unwrap();
-    }
-
-    println!("cargo:rustc-link-search=framework={}/MicrosoftCognitiveServicesSpeech.xcframework/macos-arm64_x86_64", sdk_output_dir.display());
-    println!("cargo:rustc-link-lib=framework=MicrosoftCognitiveServicesSpeech");
-
-    let inc_arg = format!("-I{}/MicrosoftCognitiveServicesSpeech.xcframework/macos-arm64_x86_64/MicrosoftCognitiveServicesSpeech.framework/Headers", sdk_output_dir.display());
-
-    let bindings_builder = bindgen::Builder::default()
-        .header("c_api/wrapper.h")
-        .clang_arg(inc_arg);
-
-    let bindings_builder = bindings_builder.clang_arg(
-        Command::new("gcc")
-            .arg("--print-file-name=include")
-            .output()
-            .map(|o| format!("-I{}", String::from_utf8_lossy(&o.stdout).trim()))
-            .unwrap(),
-    );
-
-    let bindings = bindings_builder
-        .generate()
-        .expect("Unable to generate bindings");
-
-    bindings
-        .write_to_file("src/ffi/bindings.rs")
-        .expect("Couldn't write bindings!");
 }
 
-#[cfg(target_os = "windows")]
-fn main() {
-    use std::{fs::File, io::BufReader};
-    use zip::ZipArchive;
-
-    let nuget_package_url = format!("https://www.nuget.org/api/v2/package/Microsoft.CognitiveServices.Speech/{SPEECH_SDK_VERSION}");
-
-    // Build scripts should not modify any files outside of the `OUT_DIR` directory,
-    // othersize `cargo publish` will fail.
-    let out_path = PathBuf::from(env::var("OUT_DIR").unwrap());
-
-    let mut renew = env::var("RENEW_SDK").map(|v| v == "1").unwrap_or(false);
-    let sdk_output_dir = out_path.join("sdk_output");
-    if !sdk_output_dir.exists() {
-        renew = true;
-        fs::create_dir_all(&sdk_output_dir).unwrap();
-    }
-
-    let sdk_zip_file = out_path.join(format!(
-        "microsoft.cognitiveservices.speech.{SPEECH_SDK_VERSION}.nupkg"
-    ));
-    if !sdk_zip_file.exists() {
-        download_file(nuget_package_url.as_str(), sdk_zip_file.to_str().unwrap());
+/// Download the SDK from the official Microsoft source into the given path,
+/// the file structure depends on the target OS.
+fn download_sdk(path: &Path) {
+    let zip_url = match TargetOs::get() {
+        TargetOs::Linux => {
+            let sdk_url = format!(
+                "https://csspeechstorage.blob.core.windows.net/drop/{SPEECH_SDK_VERSION}/SpeechSDK-Linux-{SPEECH_SDK_VERSION}.tar.gz");
+            let sdk = ureq::get(&sdk_url)
+                .call()
+                .expect("Failed to download Speech SDK");
+            let mut archive = tar::Archive::new(GzDecoder::new(sdk.into_reader()));
```

Comment:
on windows getting this error during build:
```

error: failed to run custom build command for `libz-ng-sys v1.1.15`                                                                                                                                      

Caused by:
  process didn't exit successfully: `C:\Users\adamb\dev\cognitive-services-speech-sdk-rs-PRs\cognitive-services-speech-sdk-rs\target\debug\build\libz-ng-sys-d764dbe13a82bc25\build-script-build_zng` (exit code: 101)
  --- stdout
  CMAKE_TOOLCHAIN_FILE_x86_64-pc-windows-msvc = None
  CMAKE_TOOLCHAIN_FILE_x86_64_pc_windows_msvc = None
  HOST_CMAKE_TOOLCHAIN_FILE = None
  CMAKE_TOOLCHAIN_FILE = None
  CMAKE_GENERATOR_x86_64-pc-windows-msvc = None
  CMAKE_GENERATOR_x86_64_pc_windows_msvc = None
  HOST_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_x86_64-pc-windows-msvc = None
  CMAKE_PREFIX_PATH_x86_64_pc_windows_msvc = None
  HOST_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = None
  CMAKE_x86_64-pc-windows-msvc = None
  CMAKE_x86_64_pc_windows_msvc = None
  HOST_CMAKE = None
  CMAKE = None
  running: "cmake" "C:\\Users\\adamb\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\libz-ng-sys-1.1.15\\src/zlib-ng" "-G" "Visual Studio 17 2022" "-Thost=x64" "-Ax64" "-DBUILD_SHARED_LIBS=OFF
" "-DZLIB_COMPAT=OFF" "-DZLIB_ENABLE_TESTS=OFF" "-DWITH_GZFILEOP=ON" "-DCMAKE_INSTALL_PREFIX=C:\\Users\\adamb\\dev\\cognitive-services-speech-sdk-rs-PRs\\cognitive-services-speech-sdk-rs\\target\\debug\
\build\\libz-ng-sys-3071debb8782d45b\\out" "-DCMAKE_C_FLAGS= -nologo -MD -Brepro" "-DCMAKE_C_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_CXX_FLAGS= -nologo -MD -Brepro" "-DCMAKE_CXX_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_ASM_FLAGS= -nologo -MD -Brepro" "-DCMAKE_ASM_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_BUILD_TYPE=Debug"

  --- stderr
  thread 'main' panicked at C:\Users\adamb\.cargo\registry\src\index.crates.io-6f17d22bba15001f\cmake-0.1.50\src\lib.rs:1098:5:

  failed to execute command: program not found
  is `cmake` not installed?

  build script failed, must exit now
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
PS C:\Users\adamb\dev\cognitive-services-speech-sdk-rs-PRs\cognitive-services-speech-sdk-rs>

```

I think this is caused by including `flate2 `library.  I know I can install `cmake `on windows but this is just pain I would like to avoid. For many users this will complicate build process significantly.

For windows we should use  `zip `package as per current version. Can this be changed accorsingly?

Reply:
Ah it looks like this was caused by me upgrading the zip package, I've turned off a whole bunch of features which should mean that it doesn't need any sys packages anymore.

Reply:
I just tested this on Windows and it means I can now build it. (and I couldn't before)